### ISSUE TITLE
Modifying `hybridGateway` permanently breaks all routes for all gateways - must uninstall

### ISSUE THREAD

**Issue by kevin-shelaga** [solo-io/gloo#5953]
### Gloo Edge Version

1.10.x (latest stable)

### Kubernetes Version

1.22.x

### Describe the bug

If you modify or remove the `hybridGateway` for any gateway, it will permanently break routing for all gateways and you must uninstall and reinstall gloo to fix routing.

### Steps to reproduce the bug

1. glooctl install gateway
2. kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo/v1.2.9/example/petstore/petstore.yaml
3. 
```
glooctl add route \
  --path-exact /all-pets \
  --dest-name default-petstore-8080 \
  --prefix-rewrite /api/pets
```
4. confirm route works - curl "$(glooctl proxy url)/all-pets"
5.
```yml
kubectl apply -n gloo-system -f - <<EOF
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: 'client-ip-reject'
  namespace: 'gloo-system'
spec:
  virtualHost:
    domains:
      - '*'
    routes:
      - matchers:
          - prefix: /
        directResponseAction:
          status: 403
          body: "client ip forbidden\n"
EOF
```
6. kubectl edit -n gloo-system gateway gateway-proxy
7. 
```yml
  hybridGateway:
    matchedGateways:
      - httpGateway:
          virtualServices:
            - name: default
              namespace: gloo-system
        matcher:
          sourcePrefixRanges:
            - addressPrefix: 1.2.3.4
              prefixLen: 32
      - httpGateway:
          virtualServices:
            - name: client-ip-reject
              namespace: gloo-system
        matcher: {}
```
8. confirm route is now blocked - curl "$(glooctl proxy url)/all-pets"
9. kubectl edit -n gloo-system gateway gateway-proxy
10. remove all `hybridGateway` config
11. confirm all routing for all gateways is now broken permanently (ex. curl "$(glooctl proxy url)/all-pets")

### Expected Behavior

Removing the hybridGateway should not force you to uninstall gloo edge completely

**Comment by gunnar-solo** [comment]
I've gone over the reproduction steps for this an _probably_ hit the same issue:

In particular:
In the initial editing of `gateway-proxy`, it is necessary to replace `httpGateway: {}` with `hybridGateway: ...`.  I found that this works as expected, as per the [documented IP blocking behavior](https://docs.solo.io/gloo-edge/master/guides/traffic_management/listener_configuration/hybrid_gateway/).  HOWEVER:  on revert, I found that I was both able to reproduce AND not reproduce the original issue.

The "on revert" part was the tricky bit.  In particular:
* directly deleting `hybridGateway: ...` caused an error
* _replacing_ `hybridGateway: ...` with `httpGateway: {}` avoided any issues

From an "envoy-config" perspective, I found that this config change resulted in the existence (or lack thereof) of a `dynamic_listener` called `"name": "listener-::-8080",`.
